### ISSUE TITLE
BITS API Example for WIndows 8.1 does not work

### ISSUE THREAD

**Issue by Nepitwin** [dahall/Vanara#257]
**Describe the bug**
I try the following bits example : https://github.com/dahall/WinClassicSamplesCS/blob/master/BacgroundIntelligenceTransferServicePolicy/TransferPolicy.cs

For newer windows version 10 and 11 both examples are working.

.Net Core 5.0 usage as example project.

By older version which use an older bits api usage the AlternateMain() methode will immediately throw an not implemented exception by the add method to create a job.

```
Creating Job...
Unhandled exception. System.NotImplementedException: The method or operation is not implemented.
  at Vanara.PInvoke.BITS.IBackgroundCopyJob.SetNotifyFlags(BG_NOTIFY NotifyFlags)
  at Vanara.IO.BackgroundCopyJob.<>c__DisplayClass134_0.<set_NotifyFlags>b__0()
  at Vanara.IO.BackgroundCopyJob.RunAction(Action action)
  at Vanara.IO.BackgroundCopyJob.set_NotifyFlags(BG_NOTIFY value)
  at Vanara.IO.BackgroundCopyJob..ctor(IBackgroundCopyJob ijob)
  at Vanara.IO.BackgroundCopyJobCollection.Add(String displayName, String description, BackgroundCopyJobType jobType)
  at Netty.Program.Main(String[] args) in C:\Users\Andreas.Sekulski\source\repos\Netty\Netty\Program.cs:line 16
```

**What code is involved**

```
var job = Vanara.IO.BackgroundCopyManager.Jobs.Add("TransferPolicy");
```

**Expected behavior**
It's not clear enough why this method is not implemented to create a job for windows 8.1.

For my opinion SetNotifyFlags COM interface should be exists and working because it's exists since windows xp. (https://docs.microsoft.com/en-us/windows/win32/api/bits/nn-bits-ibackgroundcopyjob)

Remarks from msdn by set notify flags usage

Use the SetNotifyFlags method in conjunction with the IBackgroundCopyJob::SetNotifyInterface and IBackgroundCopyJob2::SetNotifyCmdLine methods to receive event notification.

Probably a set event notification usage is missing which force this example why this does not work. But by the current state, i don't know.

**Comment by Nepitwin** [comment]
Additional information:

After small investigation the bits api version detection contains issue by detection and the Add operation is using the oldest bits api 1.0 which does not include the SetNotifyFlags implementation.

```
public static Version Version
		{
			get
			{
				try { return ver ?? (ver = GetVer()); }
				catch { return new Version(); }

				static Version GetVer()
				{
					var fi = System.Diagnostics.FileVersionInfo.GetVersionInfo(Environment.ExpandEnvironmentVariables(@"%WinDir%\Sysnative\qmgr.dll"));
					return $"{fi.FileMajorPart}.{fi.FileMinorPart}" switch
					{
						"7.8" => new Version(10, 1),
						"7.7" => new Version(5, 0),
						"7.5" => new Version(4, 0),
						"7.0" => new Version(3, 0),
						"6.7" => new Version(2, 5),
						"6.6" => new Version(2, 0),
						"6.5" => new Version(1, 5),
						"6.2" => new Version(1, 2),
						_ => new Version(1, 0),
					};
				}
			}
		}
```

**Comment by dahall** [comment]
Looking into this, the `Version` property is never called within the library so it's failure should not affect any execution.

The `IBackgroundCopyJob::SetNotifyInterface` method is called before the `IBackgroundCopyJob::SetNotifyFlags` as in the example in Microsoft Docs, so that isn't likely causing the error. The error indicates that one of the flag values passed is invalid and not supported, rather than the method. To that end, please confirm that the `Add` method only fails under Windows 8.1, but not Windows 10 or 11. That will help me research and create any conditions to prevent the failure.

**Comment by Nepitwin** [comment]
@dahall 

Thanks for you response. Yes i can confirm that the corresponding code is working fine for windows 10 and 11 machines. I can add jobs and then include download files to the job.

Only for Windows 8.1 it's failing currently.

**Comment by dahall** [comment]
If I were to send you a simple project, are you willing to compile and run it on Windows 8.1?

**Comment by dahall** [comment]
[TestBITS.zip](https://github.com/dahall/Vanara/files/7695996/TestBITS.zip)
Attached is a small solution with a simple project and simple code to test on your Win 8.1 system. I've tried to make it very readable so you can review to confirm there are no security concerns. I believe the problem is with `BG_NOTIFY_FILE_RANGES_TRANSFERRED` which I believe is Windows 10 specific. The test app will prove that out.

**Comment by dahall** [comment]
I just pushed a commit that I believe fixes both issues (failure during add and Version reporting). If you want to test this instead of building the ZIPed project above, you can pull it from the AppVeyor NuGet feed. See the homepage for details. Please let me know either way.

**Comment by Nepitwin** [comment]
Thank you, I will try both approaches on Windows 8.1 and submit feedback to you afterwards.

**Comment by Nepitwin** [comment]
Test from ZIPed project without your pushed commit.

Windows 8.1 X64
![image](https://user-images.githubusercontent.com/12249055/145796282-aa39fbf6-0586-43c5-8f51-3fdf202af40b.png)

Windows 8.1 X86
![image](https://user-images.githubusercontent.com/12249055/145796470-e3ba83d5-d53d-4d7e-928b-895a584f1775.png)

Next step is to retest your example with the additional commit

**Comment by Nepitwin** [comment]
Pre patched Vanara project 3.3.15 from appveyor nuget feed.

It's working Add Method does not throw an exception anymore.

![image](https://user-images.githubusercontent.com/12249055/145798793-818135b0-7075-4b16-8c96-412577fa9eb0.png)
